pallets/runtime/src/cdd_check.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{multisig, Runtime};
+
+use polymesh_primitives::AccountKey;
+use polymesh_runtime_common::traits::balances::CheckCdd;
+use polymesh_runtime_identity as identity;
+
+use codec::{Decode, Encode};
+use core::convert::TryFrom;
+use frame_support::StorageMap;
+
+type Identity = identity::Module<Runtime>;
+
+#[derive(Default, Encode, Decode, Clone, Eq, PartialEq)]
+pub struct CddChecker;
+
+impl CheckCdd for CddChecker {
+    fn check_key_cdd(key: &AccountKey) -> bool {
+        if let Some(did) = Identity::get_identity(&key) {
+            if Identity::has_valid_cdd(did) {
+                return true;
+            }
+        }
+        if <multisig::KeyToMultiSig<Runtime>>::contains_key(&key) {
+            let ms = <multisig::KeyToMultiSig<Runtime>>::get(&key);
+            if let Ok(ms_key) = AccountKey::try_from(ms.encode()) {
+                if let Some(did) = Identity::get_identity(&ms_key) {
+                    if Identity::has_valid_cdd(did) {
+                        return true;
+                    }
+                }
+            }
+            let ms_creator_did = <multisig::MultiSigCreator<Runtime>>::get(ms);
+            return Identity::has_valid_cdd(ms_creator_did);
```

Comment:
Trying to understand when this use-case is not captured above?
i.e. how a MS could be associated with an entity (`Identity::get_identity(&ms_key)`) which is not the creator of that multisig (<multisig::MultiSigCreator<Runtime>>::get(ms))?

Reply:
multisigs are like regular accounts. They can be added as a signer of a different identity from the creator. For example, Alice can create a multisig and Bob can add that multisig as a signer to his Identity. Once the multisig accepts the invitation, the creator of the multisig is Alice but the Identity the multisig is linked to is Bob.

Is this something that should be explicitly blocked?

Reply:
Yep - the intention is that only the identity that originally created the multisig can add it as a key (master or signing) to itself. As you say otherwise there is some ambiguity as to whether to associate its actions with the identity that created it, or the identity to which it's attached.

Would it best be tackled in a different PR?

Reply:
I think I'll make the changes to this PR itself. It will also require changes in https://github.com/PolymathNetwork/Polymesh/pull/268. I'll let you decide if this should be captured as an additional task in Jira.